Some hard-coded paths make problems when package is built into chroot like
Software Collections. Removing these hard-coded paths should fix it.

Upstream report: https://mariadb.atlassian.net/browse/MDEV-6485

diff -Naur mysql-8.0.43/cmake/install_layout.cmake mysql-8.0.43_patched/cmake/install_layout.cmake
--- mysql-8.0.43/cmake/install_layout.cmake	2025-07-09 10:45:01.000000000 +0200
+++ mysql-8.0.43_patched/cmake/install_layout.cmake	2025-08-05 09:22:54.506782584 +0200
@@ -106,7 +106,7 @@
     " Choose between ${VALID_INSTALL_LAYOUTS}" )
   ENDIF()
 
-  SET(SYSCONFDIR "${CMAKE_INSTALL_PREFIX}/etc"
+  SET(SYSCONFDIR "/etc"
     CACHE PATH "config directory (for my.cnf)")
   MARK_AS_ADVANCED(SYSCONFDIR)
 ENDIF()
@@ -202,6 +202,7 @@
 ELSE()
   SET(INSTALL_SBINDIR_RPM                 "sbin")
 ENDIF()
+SET(INSTALL_SYSCONFDIR_RPM              "/etc")
 #
 IF(CMAKE_SYSTEM_PROCESSOR IN_LIST KNOWN_64BIT_ARCHITECTURES)
   SET(INSTALL_LIBDIR_RPM                "lib64/mysql")
diff -Naur mysql-8.0.43/mysys/my_default.cc mysql-8.0.43_patched/mysys/my_default.cc
--- mysql-8.0.43/mysys/my_default.cc	2025-07-09 10:45:01.000000000 +0200
+++ mysql-8.0.43_patched/mysys/my_default.cc	2025-08-05 09:24:59.510804797 +0200
@@ -1652,12 +1652,12 @@
 
 #else
 
-  errors += add_directory(alloc, "/etc/", dirs);
-  errors += add_directory(alloc, "/etc/mysql/", dirs);
-
 #if defined(DEFAULT_SYSCONFDIR)
   if (DEFAULT_SYSCONFDIR[0])
+  {
     errors += add_directory(alloc, DEFAULT_SYSCONFDIR, dirs);
+    errors += add_directory(alloc, DEFAULT_SYSCONFDIR "/mysql", dirs);
+  }
 #endif /* DEFAULT_SYSCONFDIR */
 
 #endif
diff -Naur mysql-8.0.43/scripts/CMakeLists.txt mysql-8.0.43_patched/scripts/CMakeLists.txt
--- mysql-8.0.43/scripts/CMakeLists.txt	2025-07-09 10:45:01.000000000 +0200
+++ mysql-8.0.43_patched/scripts/CMakeLists.txt	2025-08-05 09:25:55.019369630 +0200
@@ -271,9 +271,9 @@
 ENDIF(UNIX)
 
 SET(prefix "${CMAKE_INSTALL_PREFIX}")
-SET(sysconfdir ${prefix})
+SET(sysconfdir ${SYSCONFDIR})
 SET(bindir ${prefix}/${INSTALL_BINDIR})
-SET(libexecdir ${prefix}/${INSTALL_SBINDIR})
+SET(libexecdir ${prefix}/${INSTALL_LIBEXECDIR})
 SET(datadir ${prefix}/${INSTALL_MYSQLSHAREDIR})
 SET(libsubdir  ${INSTALL_LIBDIR})
 SET(pkgincludedir ${prefix}/${INSTALL_INCLUDEDIR})
diff -Naur mysql-8.0.43/scripts/mysqld_multi.pl.in mysql-8.0.43_patched/scripts/mysqld_multi.pl.in
--- mysql-8.0.43/scripts/mysqld_multi.pl.in	2025-07-09 10:45:01.000000000 +0200
+++ mysql-8.0.43_patched/scripts/mysqld_multi.pl.in	2025-08-05 09:26:57.252077640 +0200
@@ -587,9 +587,7 @@
 
   my %seen;  # Don't list the same file more than once
   return grep { defined $_ and not $seen{$_}++ and -f $_ and -r $_ }
-              ('/etc/my.cnf',
-               '/etc/mysql/my.cnf',
-               '@sysconfdir@/my.cnf',
+              ('@sysconfdir@/my.cnf',
                ($ENV{MYSQL_HOME} ? "$ENV{MYSQL_HOME}/my.cnf" : undef),
                $opt{'extra-file'},
                ($ENV{HOME} ? "$ENV{HOME}/.my.cnf" : undef));
